custom_components/tuya_local/devices/ips_pro_heatpump.yaml

```diff
@@ -1,19 +1,54 @@
-name: IPS heatpump
+name: IPS Inverter Heatpump
 primary_entity:
   entity: climate
   dps:
     - id: 1
-      name: hvac_mode
+      name: power
```

Comment:
This change will prevent the turn_on and turn_off services from working.
It is better to keep this as hvac_mode, and make the "on" case conditional on the dp 105 (under another name)

Reply:
it is working this way, havac_mode does the value_redirect. 
Where would you place the modes then?

Reply:
Either way will work for the set_hvac_mode service, and turn_off will work as long as "off" is one of the modes. But turn_on will not keep the previous mode if the hvac_mode dp is not boolean.
So you need to turn that on its head:
```
  - id: 1
    type: boolean
    name: hvac_mode
    mapping:
        - dps_val: false
          value: "off"
        - dps_val: true
          constraint: work_mode
          conditions:
            - dps_val: smart
              value: heat_cool
            - dps_val: cool
              value: cool
            - dps_val: warm
              value: heat
```

Reply:
it's not working this way :(

Reply:
i need to check it later again, please do not merge as it is not working yet

Reply:
Thank you for your review. I have applied your suggestion, it works with my heatpump 👍